hack/tests/subcommand-scorecard.sh

```diff
@@ -16,36 +16,43 @@ set -ex
 pushd test/test-framework
 
 # test to see if scorecard fails when version is v1alpha2 and when external plugins are configured
-operator-sdk scorecard --version v1alpha2 --config "$CONFIG_PATH" |& grep '^.*error validating plugin config.*$'
+if ! $(operator-sdk scorecard --version v1alpha2 --config "$CONFIG_PATH" |& grep -q '^.*error validating plugin config.*$'); then
+	echo "expected scorecard to fail when version is v1alpha2 and when external plugins are configured"
+	exit 1
+fi
 
 # test to see if v1alpha2 is used from the command line
-commandoutput="$(operator-sdk scorecard --version v1alpha2 --config "$CONFIG_PATH_V1ALPHA2" 2>&1)"
-failCount=`echo $commandoutput | grep -o "fail" | wc -l`
-expectedFailCount=3
-if [ $failCount -ne $expectedFailCount ]
-then
-	echo "expected fail count $expectedFailCount, got $failCount"
-	exit 1
+if ! commandoutput="$(operator-sdk scorecard --version v1alpha2 --config "$CONFIG_PATH_V1ALPHA2" 2>&1)"; then 
```

Comment:
If this command exits with `0` exit code, we'll skip this block of tests. Should we fail if it exits `0`?

Similar question for the below tests as well. Should we have logic that handles the case where the command returns an unexpected exit code?

Reply:
should be fixed with this last update, the tests now only use the bash inversion (!) when the expected return code is 1, I added an 'else' to those cases if a 1 is not received. 

when the test is expected to return a 0, I let the 'set -e' do its thing and don't use bash inversion.